sdks/cpp/README.md

```diff
@@ -115,34 +115,49 @@ For more information, you can also read the [SDK Overview](../), check out [sdk.
 When running on Agones, the above functions should only fail under exceptional circumstances, so please 
 file a bug if it occurs.
 
-## Dynamic and Static Libraries
-
-In the [releases](https://github.com/googlecloudplatform/agones/releases) folder
-you will find two archives for download that contain both static and dynamic libraries for building your
-game server on Linux:
-
-- `argonsdk-$(VERSION)-dev-linux-arch_64.tar.gz`: This includes all the 
-headers as well as dynamic and static libraries that are needed for development and runtime.
-- `argonsdk-$(VERSION)-runtime-linux-arch_64.tar.gz`: This includes just the dynamic libraries that 
-are needed to run a binary compiled against the SDK and its dependencies.
-
-### Building the Libraries
+## Building From Source
+To build Agones C++ SDK from sources you need to set up your development environment and install following tools:
+- CMake (https://cmake.org/download/)
+- OpenSSL (https://github.com/openssl/openssl)
+- Git (https://git-scm.com/downloads)
+- gRPC (https://github.com/grpc/grpc)
 
-If you want to build the libraries from Agones source, 
-the `make` target `build-sdk-cpp` will compile both static and dynamic libraries for Debian/Linux
-for your usage, to be found in the `bin` directory inside this one.
+To build gRPC you may also need to install next dependencies:
+- ActivePerl (https://www.activestate.com/products/activeperl/downloads/)
+- GoLang (https://golang.org/dl/)
+- Yasm (https://yasm.tortall.net/Download.html), windows only
 
-## Building From Source
-If you wish to compile from source, you will need to compile and install the following dependencies:
+On Windows, it is possible to use Chocolatey package manager (https://chocolatey.org/docs/installation) to install dependencies:
 
-- [gRPC](https://grpc.io), v1.8.x - [C++ compilation guide](https://github.com/grpc/grpc/blob/v1.8.x/INSTALL.md)
-- [protobuf](https://developers.google.com/protocol-buffers/), v3.5.0 - [C++ compilation guide](https://github.com/google/protobuf/blob/master/src/README.md)
+    choco install git cmake activeperl golang yasm
 
+It is possible to use your own build of gRPC. In this case you need to make your **gRPC** and **protobuf** installation visible to CMake, so `find_package` command can access all necessary cmake scripts.
```

Comment:
I don't think we really want users to be doing this. They have no need to - and it's going to be a source of pain for us to maintain. We should dictate the gRPC installation version, and it's install process, and manage it for the user.

To make things simpler for everyone, I think we should remove all customisation options for being able to provide your own gRPC. Our build system should manage this for the user end to end.

Reply:
External dependencies can be kind of tricky. If a project already uses grpc (or other) you can't expect them to have 2 copies of the same library at different versions. For this reason, its generally better to allow for this customization.

Reply:
That's not to say that we cannot provide a version of grpc if one isn't specified however.

Reply:
Then we should definitely provide a default if one isn't provided 👍

I'm just very scared of trying to support someone running a random version of gRPC, and then expecting us to support it.

Reply:
Then we should definitely provide a default if one isn't provided 👍

I'm just very scared of trying to support someone running a random version of gRPC, and then expecting us to support it.

Reply:
Projects may use gRPC for their own needs and in common way it is not possible to mix different versions of gRPC in one binary.
It is possible to do it only with some restrictions: only if SDK was built as dynamic library with statically linked gRPC. It means that client will use its own instance of gRPC, and SDK will use its own.
But it is a danger way, because both instances of gRPC (application's and SDK's) may share same kernel objects (named mutex, shared memory etc.) which may cause unexpected problems. To know for sure it is necessary to know gRPC internals in depth.
My suggestion is to provide a choice to an end user how to link with gRPC. But we can show a warning if user tries to use unexpected version of gRPC to build SDK. Or even to disable the build unless user explicitly state that he wants to use different version of gRPC. In this case we can support only exact version of gRPC; in other cases user will be responsible for possible problems.

Reply:
> My suggestion is to provide a choice to an end user how to link with gRPC. But we can show a warning if user tries to use unexpected version of gRPC to build SDK. 

This sounds like the perfect option to me :+1: 

gRPC is generally SUPER backward compatible, but we should show a warning just in case.